tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -1150,40 +1251,90 @@ def parse_BoardFile(fileName):
     for EachPin in PinData:
         try:
             PinLabel[EachPin] = PinData[EachPin]["Signal"]
+            if "LED" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "BUTTON" in PinLabel[EachPin].upper():
+                BUTTON_list.append(EachPin)
         except:
             pass
 
         try:
             PinLabel[EachPin] = PinData[EachPin]["GPIO_Label"]
+            if "LED" in PinLabel[EachPin].upper():
```

Comment:
I think this can be simplified to:
```
if any(led in PinLabel[EachPin].upper() for led in ["LED", "LD1", "LD2", "LD3", "LD4", "LD5", "LD6", "LD7", "LD8", "LD9"]):
    LED_list.append(EachPin)
elif any(button in PinLabel[EachPin].upper() for button in ["BUTTON", "B_USER", "BTN"]):
    BUTTON_list.append(EachPin)
```

Reply:
👍